entities/TV.gd

```diff
@@ -0,0 +1,35 @@
+"""
+Logic of the TV
+"""
+extends "res://sources/entity/Interactable.gd"
+
+export (PackedScene) var puzzle_scene
+
+
+func _ready() -> void:
+	assert(puzzle_scene)
+
+
+func on_interacted(_actor: Node2D) -> void:
+	var puzzle = Utils.get_game().puzzle_start(puzzle_scene)
+
+	assert(puzzle.connect("puzzle_won", self, "puzzle_won") == OK)
+
+
+func on_range_entered(_actor: Node2D) -> void:
+	modulate.r = 0
+	modulate.g = 0
+
+
+func on_range_exited(_actor: Node2D) -> void:
+	modulate.r = 1
+	modulate.g = 1
+
+
+func puzzle_won() -> void:
+	"""
+	Called if player succeed in the puzzle
+	"""
+	# TODO
```

Comment:
I have idea here: TV will handle solved state of puzzle and show different sprite

Reply:
Yep, I thought of that too but we are short on assets right now so it will probably be for later if we have time, right now, a simple dialog can do the trick